aips/aip-55.md

```diff
@@ -0,0 +1,97 @@
+
+Beyond this concern, Aptos only supports a single multi-key scheme, ed25519. Multi-key schemes provide value when users can leverage different proof types for different purposes, such as leveraging Ed25519 for their wallet and a Secp256k1 from HSMs for account recovery. New technologies like Passkeys and OAuth login systems may require additional cryptographic algorithms. Combining these disparate technologies together improves user experience around managing a single account from a plethora of devices, platforms, and environments.
+
+This AIP introduces a new `TransactionAuthenticator` called `SingleSender` that supports two `AccountAuthenticator`s, SingleKeyAuthenticator and MultiKeyAuthenticator, which supports a single key and a k-of-n multi-key, respectively. These authenticators decouple the proof type from both the `TransactionAuthenticator` and `AccountAuthenticator` simplifying the addition of new cryptographic proofs for account authentication.
+
+### Goals
+
+* Eliminate duplication of code across `TransactionAuthenticator`, `AccountAuthenticator`, and `AuthencticationKey` by providing a single means to represent the identity therein.
+* Make adding new cryptographic protocols consistent across multi-sender and single-sender transactions. Current single-sender transactions expect a specific type of cryptographic algorithm in the `TransactionAuthenticator` where as multi-sender transactions expect them in the `AccountAuthenticator`.
+* Deliver a unified framework for MultiKey authentication that works across both single and multi-sender transactions.
+
+## Alternative solutions
+
+This work could have focused purely on multi-key solutions that work across the board. This would have eliminated the optimization for single-sender applications. Doing so increases the burden for creating adding cryptographic algorithms for authorization.
+
+Instead of doing this at an account level, we could have used Multisigv2. In Multisigv2, multiple on-chain accounts share a single account. This is similar to the model here, where multiple keys share the same account. The caveat being that if keys are misplaced in a Multisigv2, the assets within those accounts are lost. Those accounts require minimally gas fees, so there would be inevitable loss.
+
+## Specification
+
+### Data Structures
+
+```
+TransactionAuthenticator::SingleSender { sender: AccountAuthenticator }
+AccountAuthenticator::SingleKey { authenticator: SingleKeyAuthenticator }
+AccountAuthenticator::MultiKey { authenticator: MultiKeyAuthenticator }
+
+pub struct SingleKeyAuthenticator {
+    public_key: AnyKey,
+    signature: AnySignature,
+}
+
+pub struct MultiKeyAuthenticator {
+    public_keys: MultiKey,
```

Comment:
You need to define the `MultiKey` struct.

Also, why not replace `MultiKey` by a `Vec<AnyKey>`?